configure.ac

```diff
@@ -301,6 +302,20 @@ AM_COND_IF([ENABLE_LARGE_TILES], [
     AC_DEFINE(CONFIG_LARGE_TILES, 0, [use small tiles])
 ])
 
+## Make a guess about the source code version
+AS_IF([test -d "${srcdir}/.git"], [
+    AS_IF([which git], [
+        srcversion_string="commit: $(cd "$srcdir"; git rev-parse HEAD)"
```

Comment:
You can also use: `git -C "$srcdir" rev-parse HEAD`

Reply:
`git describe` with `--dirty` and `--broken` seems good. It can't do anything about patches added after configuration, but it will catch patches applied before config even without a commit. Changed to: `git -C "$srcdir" describe --tags --long --always --dirty --broken --abbrev=40`